iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/optional.inl

```diff
@@ -288,7 +276,7 @@ template <typename T>
 template <typename... Targs>
 inline void optional<T>::construct_value(Targs&&... args) noexcept
 {
-    new (static_cast<T*>(static_cast<void*>(&m_data))) T(std::forward<Targs>(args)...);
+    new (&m_data) T(std::forward<Targs>(args)...);
```

Comment:
Correct. New does not require a typed pointer and `m_data` is properly aligned by design.

Reply:
Thanks.